tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_NA_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta.yaml

```diff
@@ -37,3 +37,7 @@ task_run_fcst:
 task_run_post:
   NNODES_RUN_POST: 8
   PPN_RUN_POST: 12
+task_run_prdgen:
+  RUN_TASK_RUN_PRDGEN: true
+  DO_PARALLEL_PRDGEN: true
+  PPN_RUN_PRDGEN: 22
```

Comment:
I believe 22 should be assigned to `NNODES_RUN_PRDGEN` instead of `PPN_RUN_PRDGEN`

Reply:
I am not sure about the parallel prdgen code. The script  generated on the fly that `mpiexec` calls launches 22 background jobs and waits, i.e. it is already parallel. However, we call the script with `mpiexec -np 22` or RUN_CMD_PRDGEN. Wouldn't that be calling the script effectively 22x22 times?

Reply:
So the CFP module is actually what spreads out the tasks.  We define RUN_CMD_PRDGEN as "mpiexec -n $nprocs --cpu-bind core cfp".  In the ex-script, it gets called with eval $RUN_CMD_PRDGEN $CMDFILE, and cfp will execute each line of the script using 1 node.  For those 22 executions we just need 22 tasks on 1 node.

Reply:
Well this doesn't work on Hera since RUN_CMD_PRDGEN's defintion does not have "cfp". The prdgen task failed with out of memory error
```
slurmstepd: error: *** JOB 42907778 ON h33m43 CANCELLED AT 2023-03-15T10:16:19 DUE TO TIME LIMIT ***
slurmstepd: error: Detected 1786 oom-kill event(s) in StepId=42907778.batch. Some of your processes may have been killed by the cgroup out-of-memory handler.

```
So I think the RUN_CMD_PRDGEN defintion for all systems other than wcoss2 should be executed serially (mpirun -np 1) since there is no cfp there.

Reply:
Yes I agree.  The job will run more slowly as a serial job without CFP, but it should still work - the wall clock will need to be longer though.  Since the primary location this job will run is on WCOSS2 I think that is acceptable though.  If we will only use CFP on WCOSS2, then run_prdgen.local.lua files are not needed for the other supported platforms.